yunwoo/baekjoon/백트랙킹/15650-N과-M(2).js

```diff
@@ -0,0 +1,45 @@
+const readline = require('readline');
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+const input = [];
+
+rl.on('line', function (line) {
+  input.push(line);
+}).on('close', function () {
+  const [n, m] = input[0].split(' ').map(Number);
+  const checkVisits = Array.from({ length: n }, () => false);
+  const temp = [];
+  let result = '';
+
+  const dfs = (count, start) => {
+    if (count === m) {
+      result += `${temp.join(' ')}\n`;
+    } else {
+      for (let i = start; i <= n; i++) {
+        if (!checkVisits[i]) {
+          checkVisits[i] = true;
+          temp.push(i);
+
+          dfs(count + 1, i);
+
+          checkVisits[i] = false;
+          temp.pop();
+        }
+      }
+    }
+  };
+
+  dfs(0, 1);
+
+  console.log(result);
```

Comment:
```suggestion
  const [n, m] = input[0].split(" ").map(Number);
  const checkVisits = Array(n).fill(false);
  const temp = [];
  let result = "";

  const dfs = (count, start) => {
    if (count === m) {
      result += `${temp.join(" ")}\n`;
      return;
    }

    for (let i = start; i <= n; i++) {
      if (!checkVisits[i]) {
        checkVisits[i] = true;
        temp.push(i);

        dfs(count + 1, i);

        checkVisits[i] = false;
        temp.pop();
      }
    }
  };

  dfs(0, 1);

  console.log(result);
```
- 풀이잘봤씁니다~ 🙇‍♂️
- 재귀함수 사용할 때는 웬만하면 return을 이용해서 재귀 종료문을 작성해주시는 걸 추천드립니다(`무한 호출 방지`) 해당 풀이에서는 return으로 `불 필요한 블록문도 제거`할 수 있습니다.
  - 재귀를 짤 때 꿀팁: 종료 조건을 먼저 작성하고, 재귀 로직을 작성한다. 
- 블록문의 depth가 깊어질 수록 가독성이 줄어드니 해당 부분도 코드짤때 고려해주시면 좋습니다 (우아한 테크코스 과제 기준 depth3이하로만 짜야되는 요구사항이 실제로 있음)

Reply:
- 아하 다음엔 return을 적극 활용해보겠습니다!!
- 헉 우테코 그 뎁스가 이런곳에서 나오는거군요.. 기억하겠습니다 👍
- 종료조건 먼저작성.. 얼른 머리에서 이런것도 뚝딱 나왔으면 좋겠습니다🥲 코멘트 감사합니다 🙇‍♂️